python/pyarrow/public-api.pxi

```diff
@@ -97,6 +97,10 @@ cdef api object pyarrow_wrap_data_type(
         out = SparseUnionType.__new__(SparseUnionType)
     elif type.get().id() == _Type_DENSE_UNION:
         out = DenseUnionType.__new__(DenseUnionType)
+    elif type.get().id() == _Type_TIME32:
+        out = Time32Type.__new__(Time32Type)
+    elif type.get().id() == _Type_TIME64:
+        out = Time64Type.__new__(Time64Type)
```

Comment:
@jorisvandenbossche I've noticed that date32 and date64 are missing here as well. But they would require more effort because Date32Type and Date64Type don't exist. I think I would need to introduced them in `types.pxi`. I'm not sure if this is necessary though, given these types don't have a custom field (unlike Time32Type which has a `unit`).

Also should the if statement in that function be in any specific order?

Reply:
> But they would require more effort because Date32Type and Date64Type don't exist. I'm not sure if this is necessary though, given these types don't have a custom field (unlike Time32Type which has a

Yes, that's indeed because they don't have any custom logic or attributes (just like primitive numeric data types also don't have a custom subclass)

> Also should the if statement in that function be in any specific order?

I don't think so, current order looks fine